WcaOnRails/app/models/competition.rb

```diff
@@ -1737,29 +1738,33 @@ def update_persons_wcif!(wcif_persons, current_user)
         registration.update!(roles: roles)
       end
       if registration && wcif_person["assignments"]
-        registration.assignments = wcif_person["assignments"].map do |assignment_wcif|
-          existing_assignment = registration.assignments.find do |assignment|
-            assignment.wcif_equal?(assignment_wcif)
+        wcif_person["assignments"].map do |assignment_wcif|
+          schedule_activity = competition_activities.find do |competition_activity|
+            competition_activity.wcif_id == assignment_wcif["activityId"]
           end
-          if existing_assignment
-            existing_assignment
+          unless schedule_activity
+            raise WcaExceptions::BadApiParameter.new("Cannot create assignment for non-existent activity with id #{assignment_wcif["activityId"]}")
+          end
+          # We need to be very careful about how we build the assignment:
+          # providing just the registration_id or the schedule_activity_id would
+          # actually trigger a select for each validation.
+          a = registration.assignments.build(
+            schedule_activity: schedule_activity,
+            station_number: assignment_wcif["stationNumber"],
+            assignment_code: assignment_wcif["assignmentCode"],
+          )
+          if a.valid?
+            new_assignments << a.attributes
```

Comment:
Do you really have to explicitly append the attributes? I would expect Rails to be "smart" enough to handle actual model objects in the `insert_all` list

Reply:
According to the [documentation](https://apidock.com/rails/v6.0.0/ActiveRecord/Persistence/ClassMethods/insert_all) it does expect attributes; I didn't check before that PR but now I can confirm you can't just pass objects :s